RUFAS/biophysical/manure/storage/composting.py

```diff
@@ -0,0 +1,574 @@
+        """
+        self._manure_to_process = copy(self._received_manure)
+
+        manure_temperature = self._determine_outdoor_storage_temperature(
+            air_temperature=current_day_conditions.mean_air_temperature
+        )
+        storage_methane = self._calculate_composting_methane_emissions(
+            manure_temperature, self._manure_to_process.non_degradable_volatile_solids, self._composting_type
+        )
+        carbon_decomposition = self._calculate_carbon_decomposition(
+            manure_temperature,
+            self._manure_to_process.non_degradable_volatile_solids,
+            self._manure_to_process.total_solids,
+        )
+        self._apply_dry_matter_loss(storage_methane, carbon_decomposition)
+
+        storage_nitrous_oxide_N = self._calculate_nitrous_oxide_emissions(
+            FRACTION_NITROGEN_LOST_TO_DIRECT_N2O_EMISSION[self._composting_type],
+            self._manure_to_process.nitrogen,
+        )
+        storage_N_loss_from_leaching = self._calculate_nitrogen_loss_to_leaching(
+            FRACTION_NITROGEN_LOST_TO_LEACHING[self._composting_type], self._manure_to_process.nitrogen
+        )
+        storage_ammonia_N = self._calculate_composting_ammonia_emissions(
+            FRACTION_NITROGEN_LOST_TO_AMMONIA_EMISSION[self._composting_type], self._manure_to_process.nitrogen
+        )
+        self._apply_nitrogen_losses(storage_nitrous_oxide_N, storage_ammonia_N, storage_N_loss_from_leaching)
+
+        self._received_manure = copy(self._manure_to_process)
+        manure_to_return = super().process_manure(current_day_conditions, time)
+
+        data_origin_function = self.process_manure.__name__
+        simulation_day = time.simulation_day
+        self._report_processor_output(
+            "storage_methane", storage_methane, data_origin_function, MeasurementUnits.KILOGRAMS, simulation_day
+        )
+        self._report_processor_output(
+            "storage_ammonia_N",
+            storage_ammonia_N,
+            data_origin_function,
+            MeasurementUnits.KILOGRAMS,
+            simulation_day,
+        )
+        self._report_processor_output(
+            "storage_nitrous_oxide_N",
+            storage_nitrous_oxide_N,
+            data_origin_function,
+            MeasurementUnits.KILOGRAMS,
+            simulation_day,
+        )
+        self._report_processor_output(
+            "storage_N_loss_from_leaching",
+            storage_N_loss_from_leaching,
+            data_origin_function,
+            MeasurementUnits.KILOGRAMS,
+            simulation_day,
+        )
+        self._report_processor_output(
+            "carbon_decomposition",
+            carbon_decomposition,
+            data_origin_function,
+            MeasurementUnits.KILOGRAMS,
+            simulation_day,
+        )
+        self._report_manure_stream(self._stored_manure, "accumulated", simulation_day)
+        self._report_manure_stream(self._received_manure, "received", simulation_day)
```

Comment:
I think this will be an empty stream if it's called after base class's `process_manure`. You might need to make a copy of of the received manure before the call.

Reply:
line 183 sets the `_received_manure` to `_manure_to_process`, so it shouldn't be empty at this time.
But I do think that we need to clear the `_received_manure`.
```suggestion
        self._report_manure_stream(self._received_manure, "received", simulation_day)
        self._received_manure = ManureStream.make_empty_manure_stream()
```

Reply:
Doesn't the process_manure clear the _recieved_manure? and line 183 happens before the call so the _recieve_manure will be cleared again once process manure is called.

Reply:
I think Matthew's correct here that this will have been emptied by the super().process_manure() from 180. I think this should send `self._manure_to_process` to get the `received_manure` values to OM.

Reply:
Sorry, line numbers are probably meaningless at this point since I've been incorporating feedback:

I think the second line here will empty the self._received_manure before it gets to send it to OM but we can use the self._manure_to_process to get those received manure values to OM.
`
self._received_manure = copy(self._manure_to_process)
        manure_to_return = super().process_manure(current_day_conditions, time)
`

Reply:
I'm not sure if we want to send the most original received manure or the `received_manure` after processing. I think this is something we can talk about in the WT.